### PR TITLE
Use <section> element instead of div with .bs-docs.section

### PR DESCRIPTION
**Pull request by kkirsche**
Use `<section>` element instead of div with `.bs-docs.section` as I believe this is more semantically correct for how the docs are organized.

### CODE REVIEW

**Comment by patrickhlauke** [comment]
Again, there's little to no real-world benefit to doing this. No user agent does anything specifically different when encountering a ``<section>`` versus a ``<div>`` (the only theoretical difference is that a ``<section>`` would take part in the document outline algorithm...but that is not implemented in any browser http://www.paciellogroup.com/blog/2013/10/html5-document-outline/)

**Comment by patrickhlauke** [comment]
So it mostly comes down to a personal/author preference, at this stage, and what they consider more semantic (despite lack of any benefit)

**Comment by kkirsche** [comment]
While I agree with you from a practical standpoint that the actual "benefit" is nil, if I understand your point, I disagree with you strongly that there is no reason for this.

Currently, Bootstrap as an organization physically sections off each portion of the documentation's code, it is separated into sections within the affix menu, and uses section in the actual class name. In my opinion, it makes sense to use the element which was designed for the purpose of designating a section even if it provides no real world benefits. If people don't use elements for what they were made for, what's the point of having them at all?

**Comment by nextgenthemes** [comment]
Without having real proof right now, i would say there are indeed real world benefits because search engines, web scrapers, screen readers ... will benetfit from this.

A div with some class can just be anything for a robot, but if its a `<section>` it knows exactly what it is and what part of website is in it. If thats really not the case now, it will be very soon.

so +1 from me.

**Comment by patrickhlauke** [comment]
I'll leave it up to @mdo, but my position stands (and yes, the question of "what's the point" is actually a good one, which many folks have put to the WHATWG itself at various points in time)

**Comment by patrickhlauke** [comment]
> If thats really not the case now,

indeed, it's not.

> it will be very soon

we'll see...

**Comment by StevenBlack** [comment]
:+1: for <section>.  Here's why.

Trivial cases are easy to defend; it's a mistake to state "*lack of benefit*" universally applies.  Trivial cases like individual blog posts and stand-alone static pages can be built with almost anything.

In non-trivial cases, structural blocks like `article`, `section`, `nav`, `aside`, `header`, `hgroup`, and `footer`, can be extremely valuable.

Take content aggregation, for example.  Aggregation means elements are combined mechanically, either by enumerating them, or nesting them.

In nested cases you never want `section>h1` to be styled like `section>section>section>h1`.  Because you never really know, *A Priori* that an `article` containing `section`s won't be nested within another `article`'s `section`.

In summary, there are benefits to structural semantics and these should not be summarily dismissed because applications in the wild aren't all trivial.

**Comment by patrickhlauke** [comment]
Thanks for chiming in, Steven. However, you seem to have missed the part about browsers not currently supporting the theoretical document outline algorithm, and therefore that the use of just ``h1`` (rather than appropriate heading level)  is strongly discouraged (in the spec even), as otherwise assistive technology users would see/hear a completely flat document structure all made up of ``h1``s.

So, I personally prefer to use what works consistently now, rather than something that is currently broken but may in some future work.

And this is not about "trivial cases vs non-trivial cases". Bootstrap does not prescribe or ship any actual HTML. So this discussion, and the PR, are specifically about the documentation, and whether or not the authors of the documentation feel that using ``<section>`` rather than ``<div>`` is necessary or not. If you're working on some of your own content, with aggregation and what have you, feel free to use ``<section>`` if you think it'll add benefits to your specific case.

**Comment by StevenBlack** [comment]
@patrickhlauke thanks for the Bootstrap-documentation context reset; you're right.  It's hard to keep-up sometimes :smile: 

That said — now that I'm on-topic — the docs could benefit from better semantics. At the moment everything is in a flat'ish DOM without much attention to semantic detail, and inconsistencies from component to component.

I'm looking at [Components](http://getbootstrap.com/component) which is a long-long page on narrow devices. All "components" are a held in individual `.row.bs-docs-section`s wherein we have`h1' with text and examples, no `h2` levels at all, then we have `h3` "sub-sections" with examples but these are all peers within the `.bs-docs-section` non-section :-)

The image below is a small grab from the `Information - View Document Outline` in the Web Developer Toolbar browser plugin.



Each component seems to have its presentation quirks, symptomatic of the semantic discipline on the page.

**Comment by patrickhlauke** [comment]
Always happy for PRs that fix these sorts of things.

**Comment by StevenBlack** [comment]
I'm on it.

It may be because we don't have all the semantic blocs we need that, in the docs, we resort to enlisting `h3` and `h4` headings for things that are, essentially, `aside`.

**Comment by kkirsche** [comment]
:+1: best part of github. Worst thing you can get is a no but it's bringing them up and trying to fix that we all agree on I think. Then in the future of the same thing has come up its at least been discussed and there is a reason we can all refer to. Then it's up to the project owners to make the call on the ultimate needs or direction of the project and its documentation.

**Comment by mdo** [comment]
This will be nullified completely in v4—no more sections within each page. Each component will have it's own page and no `<section>`s will be needed anywhere really. I'm more inclined to see any hierarchical problems fixed before we jump into updating 60 files to simply replace an element.
